### PR TITLE
fix(swap): Return witnessedAt for boosted swaps in getStatusV2

### PR DESCRIPTION
**Pull request by niklasnatter**
Boosted swaps do not have a `witnessedAt` at the moment even when they are completed: https://chainflip-swap.chainflip.io/v2/swaps/4642972-Bitcoin-11598

### CODE REVIEW

File: packages/swap/prisma/schema.prisma [hunk 1]
```diff
@@ -139,6 +139,7 @@ model SwapRequest {
   srcAddress                   String?
   destAddress                  String?
   swapRequestedAt              DateTime
+  swapRequestedBlockIndex      String?
```
Comment (niklasnatter) [inline packages/swap/prisma/schema.prisma:142]: we use this as `witnessedAt` for vault swaps

File: packages/swap/src/routes/v2/utils.ts [hunk 2]
```diff
@@ -176,8 +176,15 @@ export const getDepositInfo = (
       witnessedAt:
         swapRequest?.originType === 'VAULT'
           ? swapRequest.swapRequestedAt.valueOf()
-          : swapRequest?.depositReceivedAt?.valueOf(),
-      witnessedBlockIndex: swapRequest?.depositReceivedBlockIndex ?? undefined,
+          : swapRequest?.depositBoostedAt?.valueOf() ??
+            swapRequest?.depositReceivedAt?.valueOf() ??
```
Comment (niklasnatter) [inline packages/swap/src/routes/v2/utils.ts:180]: a boosted swap does not have a `depositReceivedAt`

Comment (acdibble) [reply packages/swap/src/routes/v2/utils.ts:180]: probably easier to just check:

```ts
swapRequest?.depositBoostedBlockIndex ??
swapRequest?.depositReceivedBlockIndex ??
swapRequest?.swapRequestedBlockIndex
```

File: packages/swap/src/routes/v2/utils.ts [hunk 3]
```diff
@@ -176,8 +176,15 @@ export const getDepositInfo = (
       witnessedAt:
         swapRequest?.originType === 'VAULT'
```
Comment (acdibble) [inline packages/swap/src/routes/v2/utils.ts:177]: same here, we can probably just remove this condition, especially considering the current behavior of swaps being requested in the same block as they are witnessed

Comment (niklasnatter) [reply packages/swap/src/routes/v2/utils.ts]: good with be too - updated

File: packages/swap/src/routes/v2/utils.ts [hunk 4]
```diff
@@ -174,10 +174,15 @@ export const getDepositInfo = (
       txRef: depositTransactionRef,
       txConfirmations: pendingDeposit?.transactionConfirmations,
       witnessedAt:
-        swapRequest?.originType === 'VAULT'
-          ? swapRequest.swapRequestedAt.valueOf()
-          : swapRequest?.depositReceivedAt?.valueOf(),
-      witnessedBlockIndex: swapRequest?.depositReceivedBlockIndex ?? undefined,
+        swapRequest?.depositBoostedAt?.valueOf() ??
+        swapRequest?.depositReceivedAt?.valueOf() ??
+        swapRequest?.swapRequestedAt?.valueOf() ??
```
Comment (jerryafr) [inline packages/swap/src/routes/v2/utils.ts:179]: wouldn't this make a swap that's just waiting for deposit to return a wrong value?

Comment (niklasnatter) [reply packages/swap/src/routes/v2/utils.ts:179]: maybe i missunderstand the question, but if there is no deposit yet there should be no `swapRequest`

Comment (jerryafr) [reply packages/swap/src/routes/v2/utils.ts:179]: yes, so we're returning the swapRequest value as witnessed block even though there is no deposit yet, is that correct?

Comment (acdibble) [reply packages/swap/src/routes/v2/utils.ts:179]: a swap request is only created after a deposit is finalized or boosted

File: packages/swap/prisma/migrations/20241002133913_add_swap_scheduled_event/migration.sql [hunk 5]
```diff
@@ -0,0 +1,2 @@
+-- AlterTable
+ALTER TABLE "public"."SwapRequest" ADD COLUMN     "swapRequestedBlockIndex" TEXT;
```
Comment (acdibble) [inline packages/swap/prisma/migrations/20241002133913_add_swap_scheduled_event/migration.sql:2]: no backfill?

File: packages/swap/src/routes/v2/utils.ts (hunk 4 above)
Comment (jerryafr) [reply packages/swap/src/routes/v2/utils.ts:179]: ah my bad, thanks

**Review by jerryafr — APPROVED** [review]

**Pull request merged**